spec/support/shared_examples/allowed_access.rb

```diff
@@ -9,8 +9,10 @@
 # The user session is configured based on the provided user role, and the action is triggered using the appropriate HTTP method.
 RSpec.shared_examples 'allowed access' do |action, method, user_role|
   it "allows #{user_role} users to access #{action}" do
-    configure_user_session(user_role)
-    public_send(method, action, params: params_for(action))
+    resource = FactoryBot.create(model_name_underscored.to_sym)
+    configure_user_session(user_role, resource)
+    params = params_for(action, resource)
+    public_send(method, action, params:)
 
```

Comment:
This change allows us to specify the individual publication or submitter profile that we are trying to log in as owners of.  It's calling helper functions located in spec/support/helpers/access_authorization.rb to correctly set up the session variables and request parameters